modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts

```diff
@@ -165,6 +169,303 @@ const addButtons = (editor: Editor, selectionTargets: SelectionTargets, clipboar
+          }
+        },
+        {
+          type: 'menuitem',
+          text: 'Copy row',
+          icon: 'duplicate-row',
+          onAction: () => {
+            editor.execCommand('mceTableCopyRow');
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste row before',
+          icon: 'paste-row-before',
+          onAction: () => {
+            editor.execCommand('mceTablePasteRowBefore');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardRows().length === 0);
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste row after',
+          icon: 'paste-row-after',
+          onAction: () => {
+            editor.execCommand('mceTablePasteRowAfter');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardRows().length === 0);
+          }
+        },
+      ]);
+    }
+  });
+
+  editor.ui.registry.addMenuButton('colclipboardactions', {
+    icon: 'cut-column',
+    tooltip: 'Column clipboard actions',
+    fetch: (callback) => {
+      callback([
+        {
+          type: 'menuitem',
+          text: 'Cut column',
+          icon: 'cut-column',
+          onAction: () => {
+            editor.execCommand('mceTableCutCol');
+          }
+        },
+        {
+          type: 'menuitem',
+          text: 'Copy column',
+          icon: 'duplicate-column',
+          onAction: () => {
+            editor.execCommand('mceTableCopyCol');
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste column before',
+          icon: 'paste-column-before',
+          onAction: () => {
+            editor.execCommand('mceTablePasteColBefore');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardCols().length === 0);
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste column after',
+          icon: 'paste-column-after',
+          onAction: () => {
+            editor.execCommand('mceTablePasteColAfter');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardCols().length === 0);
+          }
+        },
+      ]);
+    }
+  });
+
+  const tableClassList = getTableClassList(editor);
+
+  if (tableClassList.length === 0) {
+    // eslint-disable-next-line no-console
+    console.error('Missing table class list');
+  } else {
+    editor.ui.registry.addMenuButton('tableclass', {
+      icon: 'table-classes',
+      tooltip: 'Table styles',
+      fetch: (callback) => {
+        callback(Arr.map(tableClassList, (value) => {
+          const item: Menu.ToggleMenuItemSpec = {
+            text: value.title,
+            type: 'togglemenuitem',
+            onAction: (_api: Menu.MenuItemInstanceApi) => {
```

Comment:
This type should not be needed here, as it'll be inferred from declaring the entire object a `Menu.ToggleMenuItemSpec`.